src/main/java/com/arpnetworking/tsdcore/sinks/WavefrontSink.java

```diff
@@ -66,9 +67,25 @@ public void recordAggregateData(final PeriodicData periodicData) {
             // Build the list of annotations based on cluster and white listed dimensions
             final Map<String, String> annotations = Maps.newHashMap();
             annotations.put("cluster", fqdsn.getCluster());
-            periodicData.getDimensions().entrySet().stream()
-                    .filter(entry -> _whiteListedDimensions.contains(entry.getKey()))
-                    .forEach(entry -> annotations.put(entry.getKey(), entry.getValue()));
+            if (_whiteListedDimensions.size() == 0) {
+                // Don't use the whitelist, send everything.
+                for (final Map.Entry<String, String> dimension : periodicData.getDimensions().entrySet()) {
+                    final String k = dimension.getKey();
+                    if (!k.equals(CombinedMetricData.CLUSTER_KEY)
+                            && !k.equals(CombinedMetricData.HOST_KEY)
+                            && !k.equals(CombinedMetricData.SERVICE_KEY)) {
+                        annotations.put(dimension.getKey(), dimension.getValue());
+                    }
+                }
+            } else {
```

Comment:
Hm, well, adding the optional whitelist made it a bit different, not too sure how you mean though - it reads pretty clearly to me (only because I wrote it). Do you mean, I should invert the nesting of the `for` loop and the `if` condition?